File: lena-config/src/test/java/org/riversoforion/lena/config/resolvers/EnvironmentValueResolverTest.java
/*
 * Copyright (c) 2024. Eric McIntyre / Rivers of Orion
 */
package org.riversoforion.lena.config.resolvers;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import uk.org.webcompere.systemstubs.environment.EnvironmentVariables;
import uk.org.webcompere.systemstubs.jupiter.SystemStub;
import uk.org.webcompere.systemstubs.jupiter.SystemStubsExtension;

import static org.assertj.core.api.Assertions.assertThat;

@ExtendWith(SystemStubsExtension.class)
class EnvironmentValueResolverTest {

    @SystemStub
    private final EnvironmentVariables envVars = new EnvironmentVariables();
    private final EnvironmentValueResolver resolver = new EnvironmentValueResolver();

    @DisplayName("resolveValue finds existing environment variables")
    @Test
    void resolveValue_Found() throws Exception {

        envVars.set("RIVER", "Lena")
               .and("COUNTRY", "RU")
               .execute(() -> {
                   assertThat(resolver.resolveValue("COUNTRY")).contains("RU");
                   assertThat(resolver.resolveValue("RIVER")).contains("Lena");
               });
    }

    @DisplayName("resolveValue is empty for non-existent environment variables")
    @Test
    void resolveValue_NotFound() throws Exception {

        envVars.execute(() -> {
            assertThat(resolver.resolveValue("RIVER")).isEmpty();
        });
    }
}

File: lena-config/src/jvmTest/kotlin/org/riversoforion/lena/config/resolvers/EnvironmentValueResolverTest.kt
/*
 * Copyright (c) 2024-2026. Eric McIntyre / Rivers of Orion
 */
package org.riversoforion.lena.config.resolvers

import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import uk.org.webcompere.systemstubs.environment.EnvironmentVariables
import uk.org.webcompere.systemstubs.jupiter.SystemStub
import uk.org.webcompere.systemstubs.jupiter.SystemStubsExtension
import kotlin.test.assertEquals
import kotlin.test.assertNull

@ExtendWith(SystemStubsExtension::class)
class EnvironmentValueResolverTest {

    @SystemStub
    private lateinit var envVars: EnvironmentVariables

    private val resolver = EnvironmentValueResolver()

    @Test
    fun resolveValue_Found() {
        envVars.set("RIVER", "Lena")
            .and("COUNTRY", "RU")
            .execute {
                assertEquals("RU", resolver.resolveValue("COUNTRY"))
                assertEquals("Lena", resolver.resolveValue("RIVER"))
            }
    }

    @Test
    fun resolveValue_NotFound() {
        envVars.execute {
            assertNull(resolver.resolveValue("RIVER"))
        }
    }
}
